code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -380,11 +382,8 @@ private String getInstanceName(Instance instance) {
         return data;
     }
 
-    @SuppressWarnings("unchecked")
     private Map<String, Object> getLaunchConfigDataAsMap(Service service) {
-        return (Map<String, Object>) DataAccessor.fields(service)
-                .withKey(ServiceDiscoveryConstants.FIELD_LAUNCH_CONFIG).withDefault(Collections.EMPTY_MAP)
-                .as(Map.class);
+        return getServiceDataAsMap(service);
```

Comment:
those are 2 diff methods. GetServiceData returns all services data including params defined on the service - scale, lb config, launch config - while launch config is meant to return only params specific to instance launch config.